internal/job/vault-cred-sync.go

```diff
@@ -152,3 +168,30 @@ func (v *VaultCredSync) storeCertData(ctx context.Context, vc *client.VaultClien
 	v.log.Infof("stored sync cert for %s:%s", certData.EntityName, certData.CertIndentifier)
 	return nil
 }
+func (v *VaultCredSync) storeGenericCredential(ctx context.Context, vc *client.VaultClient, secretIdentifier, secretData string) error {
+	var genericCredData GenericCredential
+	err := json.Unmarshal([]byte(secretData), &genericCredData)
+	if err != nil {
+		return errors.WithMessagef(err, "failed to parse %s secret data", secretIdentifier)
+	}
+
+	if len(genericCredData.EntityName) == 0 || len(genericCredData.CertIndentifier) == 0 || len(genericCredData.CredentialType) == 0 {
+		return errors.WithMessagef(err, "credential attributes are emty for %s secret data", secretIdentifier)
+	}
+
+	cred := map[string]string{genericCredentialClusterIdKey: genericCredData.ClusterId,
+		genericCredentialTokenKey: genericCredentialTokenKey}
+	for key, val := range genericCredData.Data {
+		cred[key] = val
+	}
+
```

Comment:
change in Generic Credential needed as commented above.
this will change after changing struct